src/planwise/component/scenarios.clj

```diff
@@ -62,7 +71,25 @@
   [store scenario-id]
   ;; TODO compute % coverage from initial scenario/projects
   (let [scenario (db-find-scenario (get-db store) {:id scenario-id})]
-    (map-scenario scenario)))
+    (when scenario (map-scenario scenario))))
+
+(defn delete-scenario
+  [store scenario-id]
+  (let [{:keys [project-id raster] :as scenario} (db-find-scenario (get-db store) {:id scenario-id})]
+    (try
+      (db-delete-scenario! (get-db store) {:id scenario-id})
+      (catch Exception e
+        (ex-info "Can not delete current scenario" {:id scenario-id} e))
+      (finally
+        (when raster
+          (let [path              (str "data/scenarios/" project-id "/")
+                scenario-filename (last (str/split raster #"/"))
+                condition-fn      (fn [file]
+                                    (when (and (.exists file) (not (.isDirectory file)))
+                                      (let [file-name (last (str/split (.getPath file) #"/"))
+                                            id        (re-find #"\d+" file-name)]
+                                        (= id (str (:id scenario))))))]
+            (delete-files-on-condition path condition-fn)))))))
```

Comment:
You should use the `scenario-filename` variable to build the names of the files to delete.